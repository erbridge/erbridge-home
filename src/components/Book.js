import PropTypes from 'prop-types';
import React, { Component, Fragment } from 'react';

import ExternalLink from './ExternalLink';

class Book extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
    authors: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string.isRequired,
      }),
    ).isRequired,
    url: PropTypes.string.isRequired,
  };

  render() {
    const { title, subtitle, authors, url } = this.props;

    return (
      <p>
        <strong>{title}</strong>
        {subtitle && (
          <Fragment>
            : <em>{subtitle}</em>
          </Fragment>
        )}{' '}
        by <strong>{authors.map(author => author.name).join(', ')}</strong> (
        <ExternalLink to={url}>Open Library</ExternalLink>)
      </p>
    );
  }
}

export default Book;
