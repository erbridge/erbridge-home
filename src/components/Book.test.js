import { shallow } from 'enzyme';
import React from 'react';

import Book from './Book';

it('renders correctly with all props', () => {
  expect(
    shallow(
      <Book
        title="Test Title"
        subtitle="Test Subtitle"
        authors={[{ name: 'Test Author' }, { name: 'Another Author' }]}
        url="http://example.com"
      />,
    ),
  ).toMatchSnapshot();
});

it('renders correctly without optional props', () => {
  expect(
    shallow(
      <Book
        title="Test Title"
        authors={[{ name: 'Test Author' }, { name: 'Another Author' }]}
        url="http://example.com"
      />,
    ),
  ).toMatchSnapshot();
});
