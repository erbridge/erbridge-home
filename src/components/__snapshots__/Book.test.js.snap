// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all props 1`] = `
<p>
  <strong>
    Test Title
  </strong>
  : 
  <em>
    Test Subtitle
  </em>
   
  by 
  <strong>
    Test Author, Another Author
  </strong>
   (
  <ExternalLink
    to="http://example.com"
  >
    Open Library
  </ExternalLink>
  )
</p>
`;

exports[`renders correctly without optional props 1`] = `
<p>
  <strong>
    Test Title
  </strong>
   
  by 
  <strong>
    Test Author, Another Author
  </strong>
   (
  <ExternalLink
    to="http://example.com"
  >
    Open Library
  </ExternalLink>
  )
</p>
`;
